FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download/DemoTasks/S3DownloadHTTPExample.c

```diff
@@ -0,0 +1,727 @@
+ *
+ * @param[in] pxTransportInterface The transport interface for making network
+ * calls.
+ * @param[in] pcPath The Request-URI to the objects of interest. This string
+ * should be null-terminated.
+ *
+ * @return The status of the file download using multiple GET requests to the
+ * server: pdPASS on success, pdFAIL on failure.
+ */
+static BaseType_t prvDownloadS3ObjectFile( const TransportInterface_t * pxTransportInterface,
+                                           const char * pcPath );
+
+/*-----------------------------------------------------------*/
+
+/*
+ * @brief Create the task that demonstrates the HTTP API Demo over a
+ * server-authenticated network connection with an HTTP server.
+ */
+void vStartSimpleHTTPDemo( void )
+{
+    /* This example uses a single application task, which in turn is used to
+     * connect, send requests, receive responses, and disconnect from the HTTP
+     * server */
+    xTaskCreate( prvHTTPDemoTask,          /* Function that implements the task. */
+                 "DemoTask",               /* Text name for the task - only used for debugging. */
+                 democonfigDEMO_STACKSIZE, /* Size of stack (in words, not bytes) to allocate for the task. */
+                 NULL,                     /* Task parameter - not used in this case. */
+                 tskIDLE_PRIORITY,         /* Task priority, must be between 0 and configMAX_PRIORITIES - 1. */
+                 NULL );                   /* Used to pass out a handle to the created task - not used in this case. */
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Entry point of the demo.
+ *
+ * This example, using a pre-signed URL,  resolves a S3 domain, establishes a
+ * TCP connection, validates the server's certificate using the root CA
+ * certificate defined in the config header, then finally performs a TLS
+ * handshake with the HTTP server so that all communication is encrypted. After
+ * which, the HTTP Client library API is used to download the S3 file (by
+ * sending multiple GET requests, filling up the response buffer each time until
+ * all parts are downloaded). If any request fails, an error code is returned.
+ *
+ * @note This example is single-threaded and uses statically allocated memory.
+ *
+ */
+static void prvHTTPDemoTask( void * pvParameters )
+{
+    /* The transport layer interface used by the HTTP Client library. */
+    TransportInterface_t xTransportInterface;
+    /* The network context for the transport layer interface. */
+    NetworkContext_t xNetworkContext = { 0 };
+    BaseType_t xIsConnectionEstablished = pdFALSE;
+    /* HTTPS Client library return status. */
+    HTTPStatus_t xHTTPStatus = HTTPSuccess;
+
+    /* Upon return, pdPASS will indicate a successful demo execution.
+    * pdFAIL will indicate some failures occurred during execution. The
+    * user of this demo must check the logs for any failure codes. */
+    BaseType_t xDemoStatus = pdPASS;
```

Comment:
```suggestion
    /*  The user of this demo must check the logs for any failure codes. */
    BaseType_t xDemoStatus = pdPASS;
```

We don't return anything in this repo